doc/specs/vulkan/chapters/VK_KHR_swapchain/wsi.txt

```diff
@@ -175,41 +176,44 @@ minimized state before a sname:VkSwapchainKHR can be created, because:
   ** If set to ename:VK_TRUE, the presentable images associated with the
      swapchain may: not own all of their pixels.
      Pixels in the presentable images that correspond to regions of the
-     target surface obscured by another window on the desktop or subject to
-     some other clipping mechanism will have undefined content when read
+     target surface obscured by another window on the desktop, or subject to
+     some other clipping mechanism, will have undefined content when read
      back.
      Pixel shaders may: not execute for these pixels, and thus any side
-     affects they would have had will not occur.
-  ** If set to ename:VK_FALSE, presentable images associated with the
-     swapchain will own all the pixels they contain.
-     Setting this value to ename:VK_TRUE does not guarantee any clipping
+     effects they would have had will not occur.
+     ename:VK_TRUE value does not guarantee any clipping
      will occur, but allows more optimal presentation methods to be used on
      some platforms.
+  ** If set to ename:VK_FALSE, presentable images associated with the
+     swapchain will own all the pixels they contain.
 
 [NOTE]
 .Note
 ====
 Applications should: set this value to ename:VK_TRUE if they do not expect
 to read back the content of presentable images before presenting them or
-after reacquiring them and if their pixel shaders do not have any side
+after reacquiring them, and if their pixel shaders do not have any side
 effects that require them to run for all pixels in the presentable image.
 ====
 
   * pname:oldSwapchain, if not dlink:VK_NULL_HANDLE, specifies an existing
-    non-retired swapchain that is associated with surface.
+    non-retired swapchain that is associated with pname:surface to indicate
+    that the application wishes to still allow use of pname:oldSwapchain's
+    images that are currently acquired.
```

Comment:
This is not the full and complete reason.  The application can: use it that way.  The original intention was so that the PE can reuse the memory of pname:oldSwapchain in creating the new swapchain.

Reply:
@ianelliottus True. I actually suspected that might be the case and use it that way.

Reply:
I don't have suggested wording changes.  I'd either remove the additional/new text, or let's figure out what to change it to.  Do you have a preference?

Reply:
Seems like a very useful information. I would prefer to add it. If not here, then below in the freeform descriptions.

Reply:
My beef with the current parameter description is that it describes validity, not semantics/behavior.

Reply:
Sounds good.  I look forward to your new wording.  Thanks!

Reply:
I dislike the proposed additional text because it is not a complete list, but as written it suggests it might be. That creates ambiguity.

A key reason for an app to use the oldSwapchain parameter is that a new swapchain cannot be created that uses the same window object. The app must use the oldSwapchain parameter, or it must destroy the old swapchain before creating a new swapchain over the same window object.  Some APIs have swapchain resize, whereas Vulkan has oldSwapchain.

Another key reason is that the memory for non-acquired images from oldSwapchain can immediately be reused by the new swapchain. This can be very important when the swapchain images consume a sizable fraction of total graphics memory.

Continued use of previously-acquired images from oldSwapchain can lead to a more glitch-free experience. It can be important too.

This passage is not the right place to enumerating all of the reasons described above, and additional reasons if there are additional reasons.